lib/Mojo/Promise.pm

```diff
@@ -290,6 +315,36 @@ reason.
     say "We are done!";
   });
 
+=head2 map
+
+  my $promise = Mojo::Promise->map(sub { ... }, @items);
+  my $promise = Mojo::Promise->map({concurrency => 3}, sub { ... }, @items);
+
+Apply a function that returns a L<Mojo::Promise> to each item in a list of
+items while optionally limiting concurrency. Returns a L<Mojo::Promise> that
+collects the results in the same manner as L</all>.
+
+Note that if any item's promise is rejected, any remaining items which have not
+yet been mapped will not be.
+
+These options are currently available:
+
+=over 2
+
+=item concurrency
+
+The maximum number of items are in progress at the same time.
+
+=back
+
+  my $promise = Mojo::Promise->map(
+    {concurrency => 3},
+    sub { $ua->get_p($_) },
+    @urls,
+  )->then(sub{
+    say $_->[0]->res->dom->at('title')->text for @_
+  });
```

Comment:
This example is out of place.

Reply:
We have methods in other classes which have a more involved example after the body of the documentation. I can't seem to find one with options too however. Where should we put such and example in that case?